internal/controller/authenticator.go

```diff
@@ -181,11 +192,23 @@ func SetUpOAuthConnection(
 	}
 	tokenSource = GetTokenSource(ctx, oauthConfig, tlsConfig)
 	creds := credentials.NewTLS(tlsConfig)
+
+	proxyDialer := func(ctx context.Context, addr string) (net.Conn, error) {
+		proxyURL, err := http.ProxyFromEnvironment(&http.Request{URL: &url.URL{Host: addr}})
+		if err != nil || proxyURL == nil {
+			logger.Warn("[Proxy Failure] No valid proxy found or error occurred", zap.Error(err))
+			return net.Dial("tcp", addr)
+		}
+		logger.Info("[Proxy Success] Using proxy", zap.String("proxy", proxyURL.String()))
```

Comment:
1. This is redundant with the logging about the HTTPS_PROXY environment variable above.
1. Remove the fancy formatting and keep it simple. When you want to add structure to log entries, use fields.
```suggestion
```